### PR TITLE
docs: decorators & error utilities PTL-1690 PTL-1692

### PR DESCRIPTION
**Pull request by SzymonZur**
ATTENTION! NEW DOCS STRUCTURE
FOR THE NEXT THREE WEEKS, ONLY BRANCH FROM new-docs-structure
OTHERWISE, WAIT UNTIL THE NEW STRUCTURE HAS BEEN RELEASED

  - Please check the [internal contributions guide](https://www.notion.so/genesisglobal/Contributing-new-documentation-75953fb245f246ff872789035451a0c4) for details
  
__________

Thank you for contributing to the documentation.

Do the changes you have made apply to both Current and Previous versions?

Have you done a trial build to check all new or changed links?

Is there anything else you would like us to know?

**This week's exciting advice from the style guide**

- Write your headings in sentence case:

  - A good example
    This is a correct heading.
  - A Bad Example
    This is not a correct heading.

### CODE REVIEW

**Review by patrickoneill-genesis — APPROVED** [review]

File: docs/001_develop/03_client-capabilities/014_utility-methods/06_decorators-util.mdx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  - css
+  - UUID
+  - unique
+  - window
+  - iframe
+---
+
+# Decorators utilities
+
+Provides utility decorators to enhance component behavior and development efficiency within TypeScript projects. Specifically, the `renderOnChange` decorator automates the re-rendering process of components upon property changes, leveraging the FAST framework for web development.
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/014_utility-methods/06_decorators-util.mdx:35]:
```suggestion
Provides utility decorators to enhance component behavior and development efficiency within TypeScript projects. Specifically, the `renderOnChange` decorator automates the re-rendering process of components upon property changes, leveraging the Genesis framework for web development.
```

File: docs/001_develop/03_client-capabilities/014_utility-methods/06_decorators-util.mdx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+## Key Features
+
+- **Automatic Rerendering:** Automatically invokes the `render()` method of a component when any decorated property changes.
+- **Integration with FAST Framework:** Designed to work seamlessly with components extending `FASTElement`.
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/014_utility-methods/06_decorators-util.mdx:40]:
```suggestion
- **Integration with Genesis Framework:** Designed to work seamlessly with components extending `GenesisElement`.
```

File: docs/001_develop/03_client-capabilities/014_utility-methods/06_decorators-util.mdx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  - serializer
+  - json
+  - styles
+  - css
+  - UUID
+  - unique
+  - window
+  - iframe
+---
+
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/014_utility-methods/06_decorators-util.mdx:32]: Can you please add a `:::info` box at the top saying decorators only apply to projects using Genesis syntax please

File: docs/001_develop/03_client-capabilities/014_utility-methods/06_decorators-util.mdx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+Provides utility decorators to enhance component behavior and development efficiency within TypeScript projects. Specifically, the `renderOnChange` decorator automates the re-rendering process of components upon property changes, leveraging the FAST framework for web development.
+
+## Key Features
+
+- **Automatic Rerendering:** Automatically invokes the `render()` method of a component when any decorated property changes.
+- **Integration with FAST Framework:** Designed to work seamlessly with components extending `FASTElement`.
+
+## Examples
+
+### Using `renderOnChange`
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/014_utility-methods/06_decorators-util.mdx:44]: Question: what is the difference between using this and an observable?

File: docs/001_develop/03_client-capabilities/014_utility-methods/06_decorators-util.mdx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+}
+```
+
+### Using with template
+
+Suppose you have a component where you want certain attributes to trigger a render when they change, such as a custom input element where the `value` and `placeholder` properties are observed:
+
+```typescript
+import { renderOnChange } from '@genesislcap/foundation-utils';
+import { attr, FASTElement, html } from '@genesislcap/web-core';
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/014_utility-methods/06_decorators-util.mdx:69]:
```suggestion
import { attr, GenesisElement, html } from '@genesislcap/web-core';
```

**Review by matteematt — CHANGES_REQUESTED** [review]
Really good. Sorry to go for the neclear response but I don't want any references to FAST merging

File: docs/001_develop/03_client-capabilities/014_utility-methods/06_decorators-util.mdx (hunk 3 above)
Comment (SzymonZur) [reply docs/001_develop/03_client-capabilities/014_utility-methods/06_decorators-util.mdx:32]: Done

File: docs/001_develop/03_client-capabilities/014_utility-methods/06_decorators-util.mdx (hunk 4 above)
Comment (SzymonZur) [reply docs/001_develop/03_client-capabilities/014_utility-methods/06_decorators-util.mdx:48]: from the PR that added this:
_renderOnChange decorator to remove some repetitive internal-observation boilerplate._

**Review by matteematt — APPROVED** [review]

**Pull request merged**